src/cljs/main/broadfcui/page/workspace/analysis/igv.cljs

```diff
@@ -62,4 +64,20 @@
        (this :refresh)))
    :refresh
    (fn [{:keys [props refs]}]
-     (.createBrowser js/igv (@refs "container") (options (:tracks props))))})
+     (let [tracks (:tracks props)]
+       (utils/log tracks)
+       (if (empty? tracks)
+         ;; if the user hasn't specified any tracks, render the IGV shell without getting a pet token
+         (.createBrowser js/igv (@refs "container") (options [] ""))
+         ;; when the user DOES have tracks, get a token for the user's pet, then pass that token into the IGV tracks.
+         (endpoints/call-ajax-sam
+           {:endpoint (endpoints/pet-token (get-in props [:workspace-id :namespace]))
+            :payload common/storage-scopes
+            :headers ajax/content-type=json
+            :on-done
+              (fn [{:keys [success? raw-response]}]
+                (if success?
+                  (let [pet-token (subs raw-response 1 (- (count raw-response) 1))]
```

Comment:
Can this string manipulation be moved to the endpoint function?  Or at least abstracted, since it's used twice.